spec/utils/uuid_utils.rb

```diff
@@ -0,0 +1,43 @@
+require 'spec_helper'
+
+describe UUIDUtils do
+
+  describe "#parse_raw and #raw" do
+
+    # Using MySQL connection (the `raw_connection`) because I didn't find a way
+    # to do prepared statements with just the `connection` object.
+    # Prepared statement seems to be the way to store binary data.
+    #
+    let(:mysql_conn) { ActiveRecord::Base.connection.raw_connection }
+
+    before(:each) do
+      mysql_conn.prepare("
+        CREATE TEMPORARY TABLE `uuid_utils_test` (
+          `id` int(11) NOT NULL AUTO_INCREMENT,
+          `uuid` binary(16) NOT NULL,
+          PRIMARY KEY (`id`)
+        )").execute
+    end
+
+    it "parses back and forth" do
+      sorted_uuids = (1..100).map { UUIDTools::UUID.timestamp_create }
```

Comment:
The timestamps are created with a 100ns granularity so if this code would happen to run faster than that you can end up with 2 or more UUIDs with the exact same timestamp component. In this case the rearranging does not specify the sort order explicitly, i.e. it's not something that matters. One option is to ensure the UUID timestamps don't overlap by adding a 1ms sleep between generating two UUIDs.